### ISSUE TITLE
Attribute parsing allows weird input with unexpected results

### ISSUE THREAD

**Issue by TedDriggs** [elichai/log-derive#3]
Consider the following code:

```rust
#[logfn(INFO, fmt = "fibonacci() -> {}", ok = "Trace", Warn)]
fn fibonacci(n: u32) -> u32 {
    match n {
        0 => 1,
        1 => 1,
        _ => fibonacci(n - 1) + fibonacci(n - 2),
    }
}
```

What level of log message will this emit?

The answer turns out to be *trace*, because the `ok = "Trace"` overrules the `INFO` and `Warn` elements, despite the return value not being a result.

Removing the `ok` clause causes these to become warnings: The `Warn` word overwrites the earlier `INFO` word.

There are a few things I was expecting to be different here.

1. Specifying the general level more than once would produce an error spanned to each subsequent attempt to redeclare. This is how `serde` handles it ([example](https://github.com/serde-rs/serde/blob/05ab569a80d2253c9106c424cd601f00615b3de8/test_suite/tests/ui/duplicate-attribute/rename-ser-rename.stderr)).
1. Attempting to use the shorthand for log-level as not the first element in the attribute would produce an error

While I get the desire to be permissive with input here, I also sort of wish the library was more particular about casing and positioning, requiring:

1. Attribute names to be snake_case to match the standard Rust style
2. Level to appear first, or possibly allowing it to appear later as `level = "warn"` with the first position being a magic shorthand.

**Comment by elichai** [comment]
I was expecting the first thing to be right too, I'll look into this.
About the second one I'm not sure if that's needed.

what exactly isn't snake_case here?
and I'm thinking maybe to throw an Error if ok/err is used on non Result function, this might be in combination with #2

**Comment by elichai** [comment]
Fixed By your PR #8
